flocker/control/test/test_httpapi.py

```diff
@@ -1309,6 +1309,263 @@ def test_container_with_volume(self):
 )
 
 
+class UpdateContainerConfigurationTestsMixin(APITestsMixin):
+    """
+    Tests for the container configuration update endpoint at
+    ``/containers/<containername>``.
+    """
+    def _create_container(self):
+        """
+        Utility function to create a container configuration via the API.
+        Also creates a second container not manipulated in the tests, to
+        ensure that unrelated applications do not get modified, moved or
+        wiped out.
+        """
+        saving = self.persistence_service.save(Deployment(
+            nodes={
+                Node(
+                    hostname=self.NODE_A,
+                    applications=[
+                        Application(
+                            name='leavemealone',
+                            image=DockerImage.from_string('busybox'),
+                        ),
+                    ]
+                ),
+                Node(hostname=self.NODE_B),
+            }
+        ))
+
+        saving.addCallback(lambda _: self.assertResponseCode(
+            b"POST", b"/configuration/containers",
+            {
+                u"host": self.NODE_A,
+                u"name": u"mycontainer",
+                u"image": u"busybox"
+            }, CREATED
+        ))
+
+        return saving
+
+    def test_update_same_host(self):
+        """
+        An API request to update a named container's host to the same host
+        on which it is already running results in an unchanged configuration.
+        """
+        d = self._create_container()
+
+        d.addCallback(lambda _: self.assertResponseCode(
+            b"POST", b"/configuration/containers/mycontainer",
+            {u"host": self.NODE_A}, OK
+        ))
+
+        def updated(_):
+            deployment = self.persistence_service.get()
+            expected = Deployment(
+                nodes={
+                    Node(
+                        hostname=self.NODE_A,
+                        applications=[
+                            Application(
+                                name='leavemealone',
+                                image=DockerImage.from_string('busybox'),
+                            ),
+                            Application(
+                                name='mycontainer',
+                                image=DockerImage.from_string('busybox')
+                            ),
+                        ]
+                    ),
+                    Node(hostname=self.NODE_B),
+                }
+            )
```

Comment:
You could probably replace this `Deployment` instantiation with a `self.persistence_service.get()` call **before** issuing the API call.  Then the assertion is more faithfully "the configuration is unchanged" because you're comparing the configuration before to the configuration afterwards.  The result is the same for now but it's less code and it's more robust to future incidental `Deployment` / configuration changes that would invalidate the assumption made here that this **particular** `Deployment` is exactly what the configuration looks like before the no-op move is made.